### ISSUE TITLE
How to understand Diffuison in this work

### ISSUE THREAD

**Issue by jeffei** [InhwanBae/SingularTrajectory#15]
Thank you so much for sharing your work on Diffusion models applied to trajectory prediction tasks!

While reading through it, I encountered a small question regarding the implementation details, and I would greatly appreciate your guidance and clarification. 

In the classic DDPM framework, the core of the model is to predict the noise $\epsilon$ added at timestep $t$ using a neural network, with the training loss typically taking the form of an MSE on the noise prediction:

<img width="325" height="48" alt="Image" src="https://github.com/user-attachments/assets/34c7b10c-61e3-42fb-a65d-da5d69ee0441" />

However, in your code implementation, I couldn't locate an explicit noise loss term in the loss function (for example, no prediction of $\epsilon_\theta$ or related MSE computation). Is this due to the adoption of a specific variant tailored for trajectory prediction?

My understanding about Diffusion method might still be limited. Could you kindly share some additional insights? This would be immensely helpful for my ongoing learning and experiments.

**Comment by InhwanBae** [comment]
Hi @jeffei,

Thank you for your keen interest in my paper and for your in-depth analysis. 

First of all, you have a perfect understanding of our work. During the development phase, I experimented with various diffusion implementations and training schemes. Finally, I found that the diffusion implementation approach used in [LED](https://github.com/MediaBrain-SJTU/LED) was the best fit for our needs. This is because the method performs denoising based on anchor-driven prototypes. Consequently, by applying the loss to the final result, I was able to get accurate samples with high fidelity.

<img width="100%" src="https://github.com/user-attachments/assets/33e8bdf0-026b-4bae-912d-528a37f2cd29" />

Here, the direct model uses the standard MSE loss for noise prediction, while ours uses the LED diffusion implementation. In addition, I recently found that the [x-loss](https://arxiv.org/abs/2511.13720), which directly predicts the next-step sample rather than the standard ϵ-loss, works better for trajectory prediction task. Hope this helps you.
